diff --git a/svf-llvm/lib/LLVMModule.cpp b/svf-llvm/lib/LLVMModule.cpp
index 749c320d..84169cf3 100644
--- a/svf-llvm/lib/LLVMModule.cpp
+++ b/svf-llvm/lib/LLVMModule.cpp
@@ -969,6 +969,10 @@ void LLVMModuleSet::buildFunToFunMap()
                     }
                 }
             }
+            // remove the app body so that the function becomes just a
+            // declartion with annotations.
+            appFuncDef->deleteBody();
+            ExtFuncsVec.push_back(appFuncDef);
         }
     }
